libs/@guardian/libs/src/measure/measure.ts

```diff
@@ -0,0 +1,49 @@
+import { isNonNullable } from '../isNonNullable/isNonNullable';
+import type { TeamName } from '../logger/@types/logger';
+import type { MeasureEnd as EndMeasure } from './@types/measure';
+
+/** For browser which do not fully support the performance API */
+const fallback = () => {
+	return -1;
+};
+
+/** Serialisation is an implementation detail */
+const SEPARATOR = ':';
+
+/**
+ * Helper to measure the duration between two events.
+ *
+ * Once ended, measures are appended to the list of `PerformanceMeasure`
+ * and can be retrieved with `performance.getEntriesByType('measure')`
+ *
+ * @example
+ * const stopMeasure = startMeasure('dotcom', 'fetch');
+ * await fetch('https://www.theguardian.com/uk.json');
+ * const duration = stopMeasure();
+ */
+export const startMeasure = (
+	team: TeamName,
+	name: string,
+	action?: string,
+): EndMeasure => {
+	if (!('getEntriesByName' in window.performance)) {
+		return fallback;
+	}
+
+	const options = {
+		start: performance.now(),
+	} satisfies PerformanceMeasureOptions;
+
+	return () => {
```

Comment:
Would this mean to stop a measure you would need to have the returned function in the same closure, stored or passed to the point where it is stopped? 

For certain scenarios that span modules or functional boundaries that might be a little cumbersome.

Could we offer another version that allows a start mark name to be passed in to stop the measure?

Reply:
I think that if we needed to cross module boundaries, we could rely in a `PerformanceObserver`. I've got a draft here: